presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java

```diff
@@ -62,6 +63,24 @@
         }
     }
 
+    protected static void assertBlockFilteredPositions(Slice[] expectedValues, Block block, List<Integer> positions)
+    {
+        Block filteredBlock = block.copyPositions(positions);
+        Slice[] filteredExpectedValues = filter(expectedValues, positions);
+        assertEquals(filteredBlock.getPositionCount(), positions.size());
+        assertBlock(filteredBlock, filteredExpectedValues);
+    }
+
+    private static Slice[] filter(Slice[] expectedValues, List<Integer> positions)
+    {
+        Slice[] prunedExpectedValues = new Slice[positions.size()];
+        int i = 0;
+        for (Integer position : positions) {
```

Comment:
This might be simpler as:
```java
for (int i = 0; i < prunedExpectedValues.length; i++) {
    prunedExpectedValues[i] = expectedValues[positions.get(i)];
} 
```